src/decorator/field-name.ts

```diff
@@ -0,0 +1,29 @@
+import { METADATA_DESERIALIZE_FIELD_NAME } from './deserialize-field-name';
+
+export const METADATA_CUSTOM_FIELDS = 'serializer:field:properties';
+/**
+ * Tags a property to map from a different field name.
+ *
+ * ## Example:
+ *
+ * ```typescript
+ * export class Example{
+ *      @FieldName('bar')
+ *      foo: string;
+ * }
+ *
+ * const result = serializer.deserialize<Example>({ bar: 'hey' }, Example);
+ * console.log(result.foo);
+ * ```
+ *
+ * This will print `hey` because the attribute with name `bar` has been mapped into the `foo` property.
+ *
+ * @decorator Property
+ */
+export function FieldName(fieldName: string): (...args: any[]) => void {
```

Comment:
The only issue doing that is that ```const customFieldNames = Reflect.getMetadata(METADATA_CUSTOM_FIELDS, target) || [];
        Reflect.defineMetadata(METADATA_CUSTOM_FIELDS, customFieldNames.concat([propertyKey]), target);``` will be called twice, but I can definetly do this.